### PR TITLE
bakerytest: new package

### PR DESCRIPTION
**Pull request by rogpeppe**
This should make it simpler to test services that issue macaroons.

Also add httpbakery.DischargeAll to make it possible to use
the httpbakery discharge machinery without going through
httpbakery.Do.

### CODE REVIEW

File: bakerytest/bakerytest.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package bakerytest
```
Comment (cmars) [inline bakerytest/bakerytest.go:3]: Add copyright.

Comment (cmars) [inline bakerytest/bakerytest.go:3]: // Package bakerytest ...

Comment (cmars) [inline bakerytest/bakerytest.go:3]: Not an issue, but questions... Is this a `bakerytest` or `httpbakerytest` package? ~~Will httpbakery get gopkg-versioned independently of bakery? If so, it seems like this package needs to get versioned along with httpbakery.~~
Edit: Looks like bakery & httpbakery get versioned together, so this is maybe ok.

File: bakerytest/bakerytest.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	d.server.Close()
+}
+
+// Location returns the location of the discharger, suitable
+// for setting as the location in a third party caveat.
+func (d *Discharger) Location() string {
+	return d.Service.Location()
+}
+
+// PublicKeyForLocation implements bakery.PublicKeyLocator
```
Comment (cmars) [inline bakerytest/bakerytest.go]: Full stop.

File: bakerytest/bakerytest.go (hunk 1 above)
Comment (rogpeppe) [inline bakerytest/bakerytest.go:3]: It's intended for testing of both httpbakery and bakery. I don't (currently) see a particular need to separate the two.

File: bakerytest/bakerytest.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+// Close when done with.
+type Discharger struct {
+	Service *bakery.Service
+
+	server *httptest.Server
+}
+
+// NewDischarger returns a new third party caveat discharger
+// which uses the given function to check caveats.
+// The cond and arg argument to the function are as returned
```
Comment (mhilton) [inline bakerytest/bakerytest.go]: s/argument/arguments/ ?

File: bakerytest/bakerytest_test.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
+package bakerytest_test
```
Comment (mhilton) [inline bakerytest/bakerytest_test.go:1]: Copyright.

**Comment by cmars** [comment]
LGTM with trivials. Can't wait to try this out.. will make mocked-up macaroon workflows even easier to read & write.

**Comment by mhilton** [comment]
:+1: LGTM

File: bakerytest/bakerytest.go (hunk 2 above)
Comment (rogpeppe) [inline bakerytest/bakerytest.go]: done

File: bakerytest/bakerytest.go (hunk 3 above)
Comment (rogpeppe) [inline bakerytest/bakerytest.go]: done

**Comment by rogpeppe** [comment]
Currently none of the bakery files have copyright messages.
I'll address that in a separate PR that adds copyright messages
to all files.

**Pull request merged**